src/validation.cpp

```diff
@@ -245,6 +245,35 @@ bool CheckFinalTx(const CTransactionRef& tx, int flags)
     return IsFinalTx(tx, nBlockHeight, nBlockTime);
 }
 
+bool GetUTXOCoin(const COutPoint& outpoint, Coin& coin)
+{
+    LOCK(cs_main);
+    if (!pcoinsTip->GetCoin(outpoint, coin))
+        return false;
+    if (coin.IsSpent())
+        return false;
+    return true;
```

Comment:
I'm not sure about this, thinking that would be better to return an `optional<Coin>`, so if the coin is spent it doesn't return the `Coin` element and it's not used by mistake by the caller.

PD: I'm just starting the review but this function smells that would be better suited for an `assertLockHeld()` rather than an internal `cs_main` lock.

Reply:
this is just a convenience function.